File: 1675. Minimize Deviation in Array/Solution.py
import heapq

class Solution:
    def minimumDeviation(self, nums: List[int]) -> int:
        pq = []
        low, res = 2e10, 2e10
        for n in nums:
            if n % 2 == 1: n <<= 1
            low = min(low, n)
            heapq.heappush(pq, -n)
        while pq:
            curr = -heapq.heappop(pq)
            res = min(res, curr - low)
            if curr % 2 == 1: break
            low = min(low, curr >> 1)
            heapq.heappush(pq, -(curr >> 1))
        return res

File: 1675. Minimize Deviation in Array/Solution.java
class Solution {
    public int minimumDeviation(int[] nums) {
        PriorityQueue<Integer> pq = new PriorityQueue<>((a, b) -> b - a);
        int low = Integer.MAX_VALUE, res = Integer.MAX_VALUE;
        for (int n : nums) {
            if (n % 2 == 1) n <<= 1;
            pq.offer(n);
            low = Math.min(low, n);
        }
        while (true) {
            int curr = pq.poll();
            res = Math.min(res, curr - low);
            if (curr % 2 == 1) break;
            low = Math.min(low, curr >> 1);
            pq.offer(curr >> 1);
        }
        return res;
    }
}